octant crash for version 0.13.0 (macOS)

**What steps did you take and what happened:**
After upgrade 0.12.1 -> 0.13.0 octant crashes when command `octant` is run.
Here are the complete logs

`2020-06-02T15:28:00.841+0100	INFO	dash/watcher.go:116	watching config file	{"component": "config-watcher", "config": "/Users/dhaven/.kube/config-dev"}
2020-06-02T15:28:00.842+0100	INFO	dash/watcher.go:116	watching config file	{"component": "config-watcher", "config": "/Users/dhaven/.kube/config-prod"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/cronJob", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/deploymentConfiguration", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/serviceEditor", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "overview/stopPortForward", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/cordon", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/uncordon", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/update", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/containerEditor", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "overview/startPortForward", "module-name": "overview"}
2020-06-02T15:28:00.843+0100	INFO	module/manager.go:79	registering action	{"component": "module-manager", "actionPath": "action.octant.dev/deleteObject", "module-name": "configuration"}
2020-06-02T15:28:02.806+0100	INFO	dash/dash.go:415	Dashboard is available at http://127.0.0.1:7777

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x15b1877]

goroutine 188 [running]:
k8s.io/apimachinery/pkg/api/resource.(*Quantity).String(0x0, 0x3d73160, 0xc0002210f0)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go:601 +0x37
github.com/vmware-tanzu/octant/internal/printer.getCombinedMetrics(0xc000385020, 0x17, 0xc0007f1700, 0xe, 0xc000fd2540, 0x10, 0x0, 0x0, 0xc000fd2560, 0xb, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/printer/horizontalpodautoscaler.go:513 +0xd5f
github.com/vmware-tanzu/octant/internal/printer.HorizontalPodAutoscalerListHandler(0x3d451c0, 0xc00235aa50, 0xc001180e70, 0x0, 0x3d67580, 0xc0010f0960, 0x3d45500, 0xc00169ee08, 0x0, 0x0, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/printer/horizontalpodautoscaler.go:47 +0x490
reflect.Value.call(0x23a6760, 0x3b45e50, 0x13, 0x25c6d84, 0x4, 0xc001a3c9f0, 0x3, 0x3, 0x100be9f, 0x2460920, ...)
	/usr/local/Cellar/go/1.14.3/libexec/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x23a6760, 0x3b45e50, 0x13, 0xc001a3c9f0, 0x3, 0x3, 0xc001a3c901, 0xc000cdb6c0, 0xc001a3c918)
	/usr/local/Cellar/go/1.14.3/libexec/src/reflect/value.go:321 +0xb4
github.com/vmware-tanzu/octant/internal/printer.(*Resource).Print(0xc001190180, 0x3d451c0, 0xc00235aa50, 0x3d126c0, 0xc001180e70, 0x3d45540, 0xc0011125a0, 0xc000cdb6c0, 0x97, 0xb, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/printer/printer.go:77 +0x472
github.com/vmware-tanzu/octant/internal/describer.(*List).Describe(0xc0022bc160, 0x3d451c0, 0xc00235aa50, 0xc0000e2433, 0xb, 0x3d67580, 0xc0010f0960, 0x3d624c0, 0xc00239f340, 0xc00235ab10, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/describer/list.go:104 +0x908
github.com/vmware-tanzu/octant/internal/describer.(*Resource).Describe(0xc00041c410, 0x3d451c0, 0xc00235aa50, 0xc0000e2433, 0xb, 0x3d67580, 0xc0010f0960, 0x3d624c0, 0xc00239f340, 0xc00235ab10, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/describer/resource.go:58 +0xd4
github.com/vmware-tanzu/octant/internal/describer.(*Section).Component(0xc00004dd00, 0x3d451c0, 0xc00235aa50, 0xc0000e2433, 0xb, 0x3d67580, 0xc0010f0960, 0x3d624c0, 0xc00239f340, 0xc00235ab10, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/describer/section.go:54 +0x58e
github.com/vmware-tanzu/octant/internal/describer.(*Section).Describe(0xc00004dd00, 0x3d451c0, 0xc00235aa50, 0xc0000e2433, 0xb, 0x3d67580, 0xc0010f0960, 0x3d624c0, 0xc00239f340, 0xc00235ab10, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/describer/section.go:36 +0xc6
github.com/vmware-tanzu/octant/internal/describer.(*Section).Component(0xc00004de00, 0x3d451c0, 0xc00235aa50, 0xc0000e2433, 0xb, 0x3d67580, 0xc0010f0960, 0x3d624c0, 0xc00239f340, 0xc00235ab10, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/describer/section.go:54 +0x58e
github.com/vmware-tanzu/octant/internal/describer.(*Section).Describe(0xc00004de00, 0x3d451c0, 0xc00235aa50, 0xc0000e2433, 0xb, 0x3d67580, 0xc0010f0960, 0x3d624c0, 0xc00239f340, 0xc00235ab10, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/describer/section.go:36 +0xc6
github.com/vmware-tanzu/octant/internal/generator.(*Generator).Generate(0xc001167a40, 0x3d451c0, 0xc00235aa50, 0xc0000e2428, 0x16, 0xc000398a78, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/generator/generator.go:107 +0x82e
github.com/vmware-tanzu/octant/internal/modules/overview.(*Overview).Content(0xc0001a78f0, 0x3d45100, 0xc001517c00, 0xc0000e2428, 0x16, 0xc000398a78, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/modules/overview/overview.go:265 +0x15a
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).generateContent(0xc0000cad20, 0x3d45100, 0xc001517c00, 0x3d660e0, 0xc0000ea420, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:162 +0x321
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).runUpdate.func1(0x3d45100, 0xc001517c00, 0x4a0d660)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:116 +0xe0
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run.func1(0x3d45100, 0xc0013c5c80, 0x3d56780, 0xc0013a2060, 0xc003313e77, 0xc00123c8a0, 0xc00036e0f0, 0xc000ed6900, 0x3b9aca00)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/api/poller.go:86 +0x271
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run(0xc0001256b0, 0x3d45100, 0xc0013c5c80, 0xc00123c8a0, 0xc000ed6900, 0x3b9aca00)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/api/poller.go:95 +0x2ff
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).Start(0xc0000cad20, 0x3d45100, 0xc0013c5c80, 0x3d660e0, 0xc0000ea420, 0x3d10a00, 0xc0010eea80)
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:106 +0x1e3
created by github.com/vmware-tanzu/octant/internal/api.(*WebsocketState).Start
	/private/tmp/octant-20200601-33665-1pevx93/src/github.com/vmware-tanzu/octant/internal/api/websocket_state.go:144 +0x9b`

**What did you expect to happen:**

octant should start up

**Anything else you would like to add:**
I use eks authentication in the kubeconfig file
aws version : aws-cli/1.17.9 Python/3.7.5 Darwin/18.7.0 botocore/1.15.1


**Environment:**

- Octant version (use `octant version`): 
0.13.0
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-13T11:51:44Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-eks-af3caf", GitCommit:"af3caf6136cd355f467083651cc1010a499f59b1", GitTreeState:"clean", BuildDate:"2020-03-27T21:51:36Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): 
MacOS Mojave 10.14.6

--- a/changelogs/unreleased/973-GuessWhoSamFoo
+++ b/changelogs/unreleased/973-GuessWhoSamFoo
@@ -1,0 +1,1 @@
+Fixed nil resource.Quantity causing panic

--- a/internal/printer/horizontalpodautoscaler.go
+++ b/internal/printer/horizontalpodautoscaler.go
@@ -495,13 +495,13 @@ func getCombinedMetrics(horizontalPodAutoscaler autoscalingv1.HorizontalPodAutos
 		switch m.Type {
 		case autoscalingv1.ObjectMetricSourceType:
 			if m.Object != nil {
-				if m.Object.MetricName != "" {
+				if m.Object.MetricName != "" && &m.Object.TargetValue != nil {
 					targets[m.Object.MetricName] = m.Object.TargetValue.String()
 				}
 			}
 		case autoscalingv1.PodsMetricSourceType:
 			if m.Pods != nil {
-				if m.Pods.MetricName != "" {
+				if m.Pods.MetricName != "" && &m.Pods.TargetAverageValue != nil {
 					target := m.Pods.TargetAverageValue.String()
 					targets[m.Pods.MetricName] = target
 				}
@@ -509,8 +509,15 @@ func getCombinedMetrics(horizontalPodAutoscaler autoscalingv1.HorizontalPodAutos
 
 		case autoscalingv1.ResourceMetricSourceType:
 			if m.Resource != nil {
-				if m.Resource.Name != "" {
+				if m.Resource.Name != "" && m.Resource.TargetAverageValue != nil {
 					targets[string(m.Resource.Name)] = m.Resource.TargetAverageValue.String()
+				}
+			}
+
+		case autoscalingv1.ExternalMetricSourceType:
+			if m.External != nil {
+				if m.External.MetricName != "" && m.External.TargetAverageValue != nil {
+					targets[string(m.External.MetricName)] = m.External.TargetAverageValue.String()
 				}
 			}
 		}
@@ -539,6 +546,12 @@ func getCombinedMetrics(horizontalPodAutoscaler autoscalingv1.HorizontalPodAutos
 			if m.Resource != nil {
 				if m.Resource.Name != "" {
 					currents[string(m.Resource.Name)] = current
+				}
+			}
+		case autoscalingv1.ExternalMetricSourceType:
+			if m.External != nil {
+				if m.External.MetricName != "" {
+					currents[string(m.External.MetricName)] = current
 				}
 			}
 		}
@@ -574,13 +587,21 @@ func getMetricStatusValue(metricStatus *autoscalingv1.MetricStatus) (string, err
 
 	switch metricStatus.Type {
 	case autoscalingv1.ObjectMetricSourceType:
-		value = metricStatus.Object.CurrentValue.String()
+		if &metricStatus.Object.CurrentValue != nil {
+			value = metricStatus.Object.CurrentValue.String()
+		}
 	case autoscalingv1.PodsMetricSourceType:
-		value = metricStatus.Pods.CurrentAverageValue.String()
+		if &metricStatus.Pods.CurrentAverageValue != nil {
+			value = metricStatus.Pods.CurrentAverageValue.String()
+		}
 	case autoscalingv1.ResourceMetricSourceType:
-		value = metricStatus.Resource.CurrentAverageValue.String()
+		if &metricStatus.Resource.CurrentAverageValue != nil {
+			value = metricStatus.Resource.CurrentAverageValue.String()
+		}
 	case autoscalingv1.ExternalMetricSourceType:
-		value = metricStatus.External.CurrentAverageValue.String()
+		if metricStatus.External.CurrentAverageValue != nil {
+			value = metricStatus.External.CurrentAverageValue.String()
+		}
 	}
 
 	return value, nil

--- a/internal/printer/horizontalpodautoscaler_test.go
+++ b/internal/printer/horizontalpodautoscaler_test.go
@@ -10,6 +10,7 @@ import (
 	"testing"
 
 	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/vmware-tanzu/octant/internal/testutil"
 	"github.com/vmware-tanzu/octant/pkg/store"
@@ -57,6 +58,11 @@ func Test_HorizontalPodAutoscalerListHandler(t *testing.T) {
 			CurrentReplicas:                 2,
 			CurrentCPUUtilizationPercentage: &currentPercentCPU,
 		},
+	}
+
+	metrics := `[{"type":"Resource","resource":{"name":"memory","targetAverageUtilization":80}}]`
+	object.Annotations = map[string]string{
+		"autoscaling.alpha.kubernetes.io/metrics": metrics,
 	}
 
 	tpo.PathForObject(object, object.Name, "/path")
@@ -360,4 +366,74 @@ func Test_createHorizontalPodAutoscalerConditionsView(t *testing.T) {
 	}...)
 
 	component.AssertEqual(t, expected, got)
+}
+
+func Test_getCombinedMetrics(t *testing.T) {
+	hpa := testutil.CreateHorizontalPodAutoscaler("hpa")
+	cases := []struct {
+		name                    string
+		horizontalPodAutoscaler autoscalingv1.HorizontalPodAutoscaler
+		annotations             map[string]string
+		expected                string
+	}{
+		{
+			name: "Object Metric Value",
+			annotations: map[string]string{
+				"autoscaling.alpha.kubernetes.io/metrics":         `[{"type":"Object","object":{"metricName":"requests-per-second","targetValue":42}}]`,
+				"autoscaling.alpha.kubernetes.io/current-metrics": `[{"type":"Object","object":{"metricName":"requests-per-second","currentValue":1}}]`,
+			},
+			horizontalPodAutoscaler: *hpa,
+			expected:                "1/42",
+		},
+		{
+			name: "Pods Metric Average Value",
+			annotations: map[string]string{
+				"autoscaling.alpha.kubernetes.io/metrics":         `[{"type":"Pods","pods":{"metricName":"packets-per-second","targetAverageValue":"350"}}]`,
+				"autoscaling.alpha.kubernetes.io/current-metrics": `[{"type":"Pods","pods":{"metricName":"packets-per-second","currentAverageValue":"1000"}}]`,
+			},
+			horizontalPodAutoscaler: *hpa,
+			expected:                "1k/350",
+		},
+		{
+			name: "Resource Metric Average Value",
+			annotations: map[string]string{
+				"autoscaling.alpha.kubernetes.io/metrics":         `[{"type":"Resource","resource":{"name":"memory","targetAverageValue":9000}}]`,
+				"autoscaling.alpha.kubernetes.io/current-metrics": `[{"type":"Resource","resource":{"name":"memory","currentAverageValue":"9001"}}]`,
+			},
+			horizontalPodAutoscaler: *hpa,
+			expected:                "9001/9k",
+		},
+		{
+			name: "Resource Metric Average Utilization",
+			annotations: map[string]string{
+				"autoscaling.alpha.kubernetes.io/metrics":         `[{"type":"Resource","resource":{"name":"memory","targetAverageUtilization":9000}}]`,
+				"autoscaling.alpha.kubernetes.io/current-metrics": `[{"type":"Resource","resource":{"name":"memory","currentAverageUtilization":9001}}]`,
+			},
+			horizontalPodAutoscaler: *hpa,
+			expected:                "",
+		},
+		{
+			name: "External Metric Average Value",
+			annotations: map[string]string{
+				"autoscaling.alpha.kubernetes.io/metrics":         `[{"type":"External","external":{"metricName":"queue_messages_ready","targetAverageValue":"0"}}]`,
+				"autoscaling.alpha.kubernetes.io/current-metrics": `[{"type":"External","external":{"metricName":"queue_messages_ready","currentAverageValue":"3"}}]`,
+			},
+			horizontalPodAutoscaler: *hpa,
+			expected:                "3/0",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.horizontalPodAutoscaler.Annotations = tc.annotations
+
+			horizontalPodAutoscalerMetrics, horizontalPodAutoscalerCurrentMetrics, err := parseAnnotations(tc.horizontalPodAutoscaler)
+			require.NoError(t, err)
+
+			result, err := getCombinedMetrics(tc.horizontalPodAutoscaler, horizontalPodAutoscalerMetrics, horizontalPodAutoscalerCurrentMetrics)
+			require.NoError(t, err)
+
+			assert.Equal(t, tc.expected, result)
+		})
+	}
 }

